utils/biospecimen.js

```diff
@@ -649,10 +659,84 @@ const biospecimenAPIs = async (req, res) => {
         if (Object.keys(query).length === 0) return res.status(404).json(getResponseJSON('Please include id to check uniqueness.', 400));
         const { checkCollectionUniqueness } = require('./firestore');
         const response = await checkCollectionUniqueness(query);
-        if (!response) return res.status(500).json({data: response, code:500});
-        return res.status(200).json({data: response, code:200})
+        if(!response) {
+            return res.status(404).json({data: response, code:404});
+        } else if (response instanceof Error) {
+           return res.status(500).json({data: response, code: 500});
+        } else {
+            return res.status(200).json({data: response, code:200});
+        }
```

Comment:
Please consider the same as above for structuring the error handling flow in the endpoint and the function it calls.